builtin/providers/nsone/resource_record_test.go

```diff
@@ -0,0 +1,296 @@
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckRecordExists(n string, record *nsone.Record) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("NoID is set")
+		}
+
+		client := testAccProvider.Meta().(*nsone.APIClient)
+
+		p := rs.Primary
+
+		foundRecord, err := client.GetRecord(p.Attributes["zone"], p.Attributes["domain"], p.Attributes["type"])
+
+		if err != nil {
+			// return err
+			return fmt.Errorf("Record not found")
+		}
+
+		if foundRecord.Domain != p.Attributes["domain"] {
+			return fmt.Errorf("Record not found")
+		}
+
+		*record = *foundRecord
+
+		return nil
+	}
+}
+
+func testAccCheckRecordDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*nsone.APIClient)
+
+	var recordDomain string
+	var recordZone string
+	var recordType string
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "nsone_record" {
+			continue
+		}
+
+		if rs.Type == "nsone_record" {
+			recordType = rs.Primary.Attributes["type"]
+			recordDomain = rs.Primary.Attributes["domain"]
+			recordZone = rs.Primary.Attributes["zone"]
+		}
+	}
+
+	foundRecord, _ := client.GetRecord(recordDomain, recordZone, recordType)
+
+	if foundRecord.Id != "" {
+		return fmt.Errorf("Record still exists")
+	}
+
+	return nil
+}
+
+func testAccCheckRecordAttributes(record *nsone.Record) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		if record.Ttl != 60 {
+			return fmt.Errorf("Bad value : %d", record.Ttl)
+		}
+
+		recordAnswer := record.Answers[0]
+		recordAnswerString := recordAnswer.Answer[0]
+
+		if recordAnswerString != "test1.terraform.io" {
+			return fmt.Errorf("Bad value : %s", record.Ttl)
+		}
+
+		if recordAnswer.Region != "cal" {
+			return fmt.Errorf("Bad value : %s", recordAnswer.Region)
+		}
+
+		recordMetas := recordAnswer.Meta
+
+		if recordMetas["weight"].(float64) != 10 {
+			return fmt.Errorf("Bad value : %b", recordMetas["weight"].(float64))
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckRecordAttributesUpdated(record *nsone.Record) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		if record.Ttl != 120 {
+			return fmt.Errorf("Bad value : %s", record.Ttl)
+		}
+
+		recordAnswer := record.Answers[1]
+		recordAnswerString := recordAnswer.Answer[0]
+
+		if recordAnswerString != "test3.terraform.io" {
+			return fmt.Errorf("Bad value for updated record: %s", recordAnswerString)
+		}
+
+		if recordAnswer.Region != "wa" {
+			return fmt.Errorf("Bad value : %s", recordAnswer.Region)
+		}
+
+		recordMetas := recordAnswer.Meta
+
+		if recordMetas["weight"].(float64) != 5 {
+			return fmt.Errorf("Bad value : %b", recordMetas["weight"].(float64))
+		}
+
+		return nil
+	}
+}
+
+const testAccRecordBasic = `
+resource "nsone_record" "foobar" {
+    zone = "terraform.io"
```

Comment:
This test is failing because the zone doesn't exist. We want to add a dependency from the `nsone_record` on the `nsone_zone`. Easiest is probably changing this line to:
```
    zone = "${nsone_zone.test.zone}"
```

that fixes the test run on my machine